py/file.go

```diff
@@ -10,17 +10,126 @@
 package py
 
 import (
+	"io"
+	"io/ioutil"
 	"os"
 )
 
-var FileType = NewTypeX("file", `represents an open file`,
-	nil, nil)
-
 type File os.File
 
+var FileType = NewType("file", `represents an open file`)
+
+func init() {
+	FileType.Dict["write"] = MustNewMethod("write", func(self Object, value Object) (Object, error) {
+		return self.(*File).Write(value)
+	}, 0, "write(arg) -> writes the contents of arg to the file, returning the number of characters written.")
+
+	FileType.Dict["read"] = MustNewMethod("read", func(self Object, args Tuple, kwargs StringDict) (Object, error) {
+		return self.(*File).Read(args, kwargs)
+	}, 0, "read([size]) -> read at most size bytes, returned as a string.\n\nIf the size argument is negative or omitted, read until EOF is reached.\nNotice that when in non-blocking mode, less data than what was requested\nmay be returned, even if no size parameter was given.")
+	FileType.Dict["close"] = MustNewMethod("close", func(self Object) (Object, error) {
+		return self.(*File).Close()
+	}, 0, "close() -> None or (perhaps) an integer.  Close the file.\n\nSets data attribute .closed to True.  A closed file cannot be used for\nfurther I/O operations.  close() may be called more than once without\nerror.  Some kinds of file objects (for example, opened by popen())\nmay return an exit status upon closing.")
+}
+
 // Type of this object
 func (o *File) Type() *Type {
 	return FileType
 }
 
+func (o *File) Write(value Object) (Object, error) {
+	var b []byte
+
+	switch v := value.(type) {
+	// FIXME Bytearray
+	case Bytes:
+		b = v
+
+	case String:
+		b = []byte(v)
+
+	default:
+		return nil, ExceptionNewf(TypeError, "expected a string or other character buffer object")
+	}
+
+	n, err := (*os.File)(o).Write(b)
+	return Int(n), err
+}
+
+func (o *File) readResult(b []byte) (Object, error) {
+	var asBytes = false // default mode is "as string" - also move this in File struct
```

Comment:
perhaps we could already do that as part of this CL ?

Reply:
Ok, I have this part done (but I am still not done parsing the options for open() so it would still just work for string output).

Found some problems in the option parser that I am trying to decide if I want to ignore them for now or trying to get them fixed.